### PR TITLE
Add API to view FxOS Add-ons in the review queue (bug 1196225)

### PR DESCRIPTION
**Pull request by diox**
https://bugzilla.mozilla.org/show_bug.cgi?id=1196234

Bonus: also start implementing publish/reject APIs (bug 1196234)

### CODE REVIEW

File: docs/api/topics/firefox_os_addons_review.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+Firefox OS Add-ons Review
+=========================
+
+.. warning::
+
+    Firefox OS Add-ons in Marketplace are experimental and not yet available in
+    production. This API is not ready for public consumption yet and can change
+    at any moment.
+
+Add-ons queue
```
Comment (ngokevin) [inline docs/api/topics/firefox_os_addons_review.rst]: capitalize headers

File: docs/api/topics/firefox_os_addons_review.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+=============
+
+.. http:get:: /api/v2/extension/queue/
```
Comment (ngokevin) [inline docs/api/topics/firefox_os_addons_review.rst]: extensions

File: docs/api/topics/firefox_os_addons_review.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    **Response**
+
+    :status 202: successfully published.
+    :status 403: not allowed to access this object.
+    :status 404: add-on not found in the review queue.
+
+.. http:post:: /api/v2/extensions/queue/(int:id)|(string:slug)/reject
+
+    Reject an add-on. Its status will be updated to "rejected". The developer
+    will have to re-submit it once he fixes the issues.
```
Comment (ngokevin) [inline docs/api/topics/firefox_os_addons_review.rst]: s / :man: / :man: :woman: /

File: docs/api/topics/firefox_os_addons_review.rst [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    **WORK IN PROGRESS** Since versionning and re-submitting a rejected add-on
```
Comment (ngokevin) [inline docs/api/topics/firefox_os_addons_review.rst]: s/versionning/versioning

**Comment by diox** [comment]
Thanks, docs are fixed now.

File: mkt/extensions/views.py [hunk 5]
```diff
@@ -152,3 +155,33 @@ def as_view(cls, **kwargs):
...
         return non_atomic_requests(view)
+
+
+class ReviewersExtensionViewSet(CORSMixin, SlugOrIdMixin, MarketplaceView,
+                                ListModelMixin, RetrieveModelMixin,
+                                GenericViewSet):
+    authentication_classes = [RestOAuthAuthentication,
+                              RestSharedSecretAuthentication,
+                              RestAnonymousAuthentication]
+    cors_allowed_methods = ('get',)
```
Comment (ngokevin) [inline mkt/extensions/views.py]: post?

**Comment by ngokevin** [comment]
r+wc/t

File: mkt/extensions/views.py (hunk 5 above)
Comment (diox) [inline mkt/extensions/views.py]: fixed

**Comment by diox** [comment]
Added CORS and CORS tests

**Pull request merged**